samza-sql/src/main/java/org/apache/samza/sql/avro/AvroRelConverter.java

```diff
@@ -125,11 +125,18 @@ public static void fetchFieldNamesAndValuesFromIndexedRecord(IndexedRecord recor
         .collect(Collectors.toList()));
   }
 
-  private static SamzaSqlRelRecord convertToRelRecord(IndexedRecord avroRecord, Schema schema) {
+  private static SamzaSqlRelRecord convertToRelRecord(IndexedRecord avroRecord) {
     List<Object> fieldValues = new ArrayList<>();
     List<String> fieldNames = new ArrayList<>();
     if (avroRecord != null) {
-      fetchFieldNamesAndValuesFromIndexedRecord(avroRecord, fieldNames, fieldValues, schema);
+      fieldNames.addAll(
```

Comment:
Can we add a simple unit test or at least a comment to avoid that this happens again in the future ?

Reply:
Done. Added unit tests for Avro record conversion with non-nullable union type.